src/Google/GlobalSiteTag.php

```diff
@@ -155,13 +198,215 @@ public function maybe_display_event_snippet( string $ads_conversion_id, string $
+			esc_js( $product->get_name() ),
+			esc_js( join( ' & ', $this->product_helper->get_categories( $product ) ) ),
+		);
+		wp_print_inline_script_tag( $view_item_gtag );
+	}
+
+	/**
+	 * Display the JavaScript code to track all pages.
+	 */
+	private function display_page_view_event_snippet(): void {
+		$page_view_gtag = sprintf(
+			'gtag("event", "page_view", {
+			send_to: "GLA",
+			developer_id.%s: "true",});',
+			esc_js( self::DEVELOPER_ID )
+		);
+		if ( ! is_cart() ) {
+			wp_print_inline_script_tag( $page_view_gtag );
+		} else {
+			// display the JavaScript code to track the cart page
+			$item_info = '';
+
+			foreach ( WC()->cart->get_cart() as $cart_item ) {
+				// gets the product id
+				$id = $cart_item['product_id'];
+
+				// gets the product object
+				$product = $cart_item['data'];
+				$name    = $product->get_name();
+				$price   = $product->get_price();
+				// gets the cart item quantity
+				$quantity = $cart_item['quantity'];
+
+				$item_info = $item_info . sprintf(
+					'{
+					id: "gla_%s",
+					price: %s,
+					google_business_vertical: "retail",
+					item_name:"%s",
+					quantity:"%s",
+					}',
+					esc_js( $id ),
+					esc_js( $price ),
+					esc_js( $name ),
+					esc_js( $quantity )
+				);
+			}
+			$value          = WC()->cart->total;
+			$page_view_gtag = sprintf(
+				'gtag("event", "page_view", {
+				send_to: "GLA",
+				ecomm_pagetype: "cart",
+				value: "%s",
+				items: [%s]});',
+				esc_js( $value ),
+				esc_js( $item_info ),
+			);
+			wp_print_inline_script_tag( $page_view_gtag );
+		}
+	}
+
+	/**
+	 * TODO : Track "add to cart button" on single pages, archive pages, and
+	 * through various shortcodes and blocks with a compatible solution.
+	 *
+	 * Display the JavaScript code to track the add to cart button.
+	 *
+	 * @param string $message Add to cart messages.
+	 * @param array  $products Product ID list.
+	 */
+	private function custom_action_add_to_cart( $message, $products ) {
+		// Only display this tag info after click the add to cart button .
+		foreach ( $products as $product_id => $value ) {
+
+			$product = wc_get_product( $product_id );
+
+			add_action(
+				'wp_footer',
+				function () use ( $product ) {
+					$add_to_cart_gtag = sprintf(
+						'gtag("event", "add_to_cart", {
+						send_to: "GLA",
+						developer_id.%s: "true",
+						ecomm_pagetype: "cart",
+						value: "%s",
+						items:[{
+							id: "gla_%s",
+							price: %s,
+							google_business_vertical: "retail",
+							item_name:"%s",
+							category:"%s",
+							}]});',
+						esc_js( self::DEVELOPER_ID ),
+						esc_js( (string) $product->get_price() ),
+						esc_js( $product->get_id() ),
+						esc_js( (string) $product->get_price() ),
+						esc_js( $product->get_name() ),
+						esc_js( join( ' & ', $this->product_helper->get_categories( $product ) ) ),
+					);
+					wp_print_inline_script_tag( $add_to_cart_gtag );
+				}
+			);
+
+			return $message;
```

Comment:
The return is done within the for loop, which means it will only track the first item added to cart.

Reply:
Good catch. Updated.